specification/2.0/appendix/performance-metrics/README.md

```diff
@@ -0,0 +1,107 @@
+# Runtime Performance Metrics 
+
+The goal of these metrics is to assess the runtime performance needs of glTF assets.  
+By runtime we mean the cpu/gpu, bandwidth and memory resources needed to render the model in realtime.  
+These performance metrics are divided in two major categories - `complexity` and `memory`
+
+## Performance Metric Categories
+
+### Complexity  
+
+The `complexity`category provides metrics to calculate cpu/gpu and memory bandwidth target requirements - larger numbers will mean a more powerful target device is needed.  
+These metrics do not provide a mapping to target device performance, this mapping must be done by users of the metrics.  
+
+### Screen space coverage
+
+The metrics do not include screen space coverage, it is advised that clients shall know how large the render area will be before requesting a model (from the backend)  
```

Comment:
Should we exclude the 3d bounds (screen space coverage) completely, though?  I would suggest revisiting this a bit.  
For AR/VR experiences, the unit of meters (though get scaled in practice, for good reasons) provide valuable information about the model.  Coupling the unit with the 3d bounds of the asset, would help set a guideline for the Spatial FavIcon creators, I believe.  
Its hard to predict, but over time, it appears that the industry would rely on the 3d bounds (along with other metrics already listed here) also for spelling out the intent of the asset.   
Curious to hear others perspectives on this.

Reply:
Ah, ok I see your point @jeymichael
I do think there is a distinction between screen space coverage and 3d boundingbox - the latter is useful for calculating the asset size as compared to rest of the 'world'.

I have assumed that traversing the nodes and calculating the 3d bounds from positional min/max is super easy - however I guess this extension is a relavant place for it.

- Would a resulting box calculated from concatenating node matrices and positional mix/min be enough?
Ie, start in scene with identity matrix, for each node cancat node transform or matrix and apply to min/max - then update bounds min/max.
This would give the outer boundingbox, in meters, for the world space transformed meshes.

Reply:
I've been assuming that these performance metrics are meant for choosing among models that are — other than some quality vs. performance tradeoffs — functionally equivalent within the application. While engines use 3D bounds for many things, does that fit here? Given an asset with world bounds of 100km and an asset of 0.01km, how would an engine choose which to show? I don't know that this document should get into defining functional and contextual differences between assets.

Reply:
I agree with what you are saying for a _normal_ usecase @donmccurdy - but I interpret the need to be for AR/VR situation.
The reason this is important is because you cannot beforehand say that the rendertarget area is X * Y pixels - that you would normally use for rendering a model in a 2D canvas.

Instead the model dimension could be used in relation to the total viewing space, the objects distance and clip regions (ie at what distance does the object get clipped) - for instance an object with (real) world dimensions of 0.5,0.5,0.5 and a clip distance of 0.3 meters would potentially take up a large portion of the screen area.
But a 0.05, 0.05, 0.05 object would only cover a fraction of the screen area and possibly being rendered with more fidelity.

This is my guess anyway :-)

Reply:
> I've been assuming that these performance metrics are meant for choosing among models that are — other than some quality vs. performance tradeoffs — functionally equivalent within the application. While engines use 3D bounds for many things, does that fit here? Given an asset with world bounds of 100km and an asset of 0.01km, how would an engine choose which to show?

The 3D Bounds proposal is not meant for assisting engines in many things, but only to add direct/relevant information about the space the asset occupies (for the AR/VR use-cases)  After going back to the original [Issue](https://github.com/KhronosGroup/glTF/issues/1743) with FavIcons/ManifestIcons that started this, I am convinced with your parallel to [Responsive Images](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images).   
With that in mind, I could see how browser can pick between different assets based on the bounds.  I believe Ravi had some screenshots / examples, for AR/VR W3C use-cases.
 
In short, 3D Bounds help in addressing the **art direction** & **resolution switching** problem.

Reply:
I had discussed with W3C community to just use "static bounding box" as analogous to "resolution" in 2D space. Though as a stand alone metric, it was not enough. Here are my analysis: https://github.com/immersive-web/spatial-favicons/blob/master/design_docs/SpatialFaviconSizes.pdf

Reply:
Thanks @raviramachandra
Looking through the document leads me to believe that a static (non animated) bounding volume for the aggregated min/max (transformed while traversing the nodetree) primitive values would do the trick?

I will add this as a metric on each scene.